From 8a6b1b3627992a9dc47e7eb419ae2f762ddeccb0 Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Tue, 27 May 2025 09:49:12 +0300
Subject: [PATCH] [SW-229822] g2e-oam: update hbm rl reset_val

After fine-tuning max_sat, reset_val and max_timeout HBM RL configuration
values, for G2E-OAM the reset_val should be updated to 51. All other
values should be common to G2E-OAM and G2E-PCIe.

Change-Id: Ia33bd1a1e906ee83accd78d5544ad138f3e1441d
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/arch/gaudi2/device.c b/arch/gaudi2/device.c
index 44f41cb..ff720ca 100644
--- a/arch/gaudi2/device.c
+++ b/arch/gaudi2/device.c
@@ -1332,7 +1332,8 @@
 #define GAUDI2E_HBM_RL_EN_SHIFT			24		/* RL enable */
 #define GAUDI2E_HBM_RL_EN_MASK			0x1000000	/* Bit 24 */
 #define GAUDI2E_HBM_RL_MAX_SAT			7
-#define GAUDI2E_HBM_RL_RESET_VAL		52
+#define GAUDI2E_HBM_RL_RESET_VAL_PCI		52
+#define GAUDI2E_HBM_RL_RESET_VAL_OAM		51
 #define GAUDI2E_HBM_RL_MAX_TIMEOUT		99
 #define GAUDI2E_HBM_RL_DCORE_NUM		4
 #define GAUDI2E_HBM_RL_RTR_NUM			8
@@ -1348,6 +1349,11 @@
 	uint32_t rd_rl_mem;
 };
 
+static int32_t hl_get_hbm_rl_reset_val(void)
+{
+	return hl_is_card_type_pcie() ? GAUDI2E_HBM_RL_RESET_VAL_PCI : GAUDI2E_HBM_RL_RESET_VAL_OAM;
+}
+
 void hl_rtr_ctrl_init_rl_reg_once(void)
 {
 	int32_t max_sat, reset_val, max_timeout, dcore, rtr;
@@ -1355,7 +1361,7 @@
 	uint32_t cfg = 0;
 
 	max_sat = GAUDI2E_HBM_RL_MAX_SAT;
-	reset_val = GAUDI2E_HBM_RL_RESET_VAL;
+	reset_val = hl_get_hbm_rl_reset_val();
 	max_timeout = GAUDI2E_HBM_RL_MAX_TIMEOUT;
 
 	cfg |= (max_sat << GAUDI2E_HBM_RL_SATURATION_SHIFT);
